.travis.yml

```diff
@@ -8,22 +41,61 @@ os:
   - linux
   - osx
 
+matrix:
+  exclude:
+    - compiler: clang
+      env: BUILD_TARGET="cc2538"
+    - compiler: clang
+      env: BUILD_TARGET="pretty-check"
+
 before_install:
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install python-pexpect; fi
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then sudo easy_install pexpect; fi
 
 install:
-  - pushd ~
-  - wget https://sourceforge.net/projects/astyle/files/astyle/astyle%202.05.1/astyle_2.05.1_linux.tar.gz/download -O astyle.tar.gz
-  - tar -xzvf astyle.tar.gz
-  - pushd astyle/build/gcc && LDFLAGS=" " make
-  - export PATH=$PWD/bin:$PATH
-  - popd
-  - popd
+  - if [ "$BUILD_TARGET" == "pretty-check" ]; then
+      pushd ~ &&
+      wget https://sourceforge.net/projects/astyle/files/astyle/astyle%202.05.1/astyle_2.05.1_linux.tar.gz/download -O astyle.tar.gz &&
+      tar xzvf astyle.tar.gz &&
+      pushd astyle/build/gcc && LDFLAGS=" " make &&
+      export PATH=$PWD/bin:$PATH &&
+      popd &&
+      popd &&
+      astyle --version ;
+    fi
+
+  - if [ "$BUILD_TARGET" == "cc2538" ] && [ "$TRAVIS_OS_NAME" == "linux" ] ; then
+      sudo apt-get install lib32z1 &&
+      pushd ~ &&
+      wget https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update/+download/gcc-arm-none-eabi-4_9-2015q3-20150921-linux.tar.bz2 &&
+      tar xjf gcc-arm-none-eabi-4_9-2015q3-20150921-linux.tar.bz2 &&
+      export PATH=$PWD/gcc-arm-none-eabi-4_9-2015q3/bin:$PATH &&
+      popd &&
+      arm-none-eabi-gcc --version ;
+    fi
+
+  - if [ "$BUILD_TARGET" == "cc2538" ] && [ "$TRAVIS_OS_NAME" == "osx" ] ; then
+      pushd ~ &&
+      wget https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update/+download/gcc-arm-none-eabi-4_9-2015q3-20150921-mac.tar.bz2 &&
+      tar xjf gcc-arm-none-eabi-4_9-2015q3-20150921-mac.tar.bz2 &&
+      export PATH=$PWD/gcc-arm-none-eabi-4_9-2015q3/bin:$PATH &&
+      popd &&
+      arm-none-eabi-gcc --version ;
+    fi
```

Comment:
It may make sense at this point to add a `.travis/` directory containing these scripts (like I do with wpantund) and have the `.travis.yml` file call into them. Might be easier to maintain.